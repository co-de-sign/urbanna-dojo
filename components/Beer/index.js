import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

// formato de uma Beer:
// beer.name => nome da cerveja
// beer.alcohol => porcentagem de alcool da cerveja
// beer.type => array de strings que contem os tipos da cerveja

export default class Beer extends Component {
  render() {
    return (
      <View style={styles.beer}>
        <Text style={styles.name}>{this.props.beer.name}</Text>
        <Text>{this.props.beer.alcohol}%</Text>
        {this.props.beer.type.map(type => (
          <Text key={type}>{type}</Text>
        ))}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  beer: {
    width: 256,
    height: 244,
    backgroundColor: '#fff',
    marginTop: 32,
    padding: 24,
  },
  name: {
    fontSize: 24,
    fontWeight: 'bold',
  }
})
