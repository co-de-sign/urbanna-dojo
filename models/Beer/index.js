export default class Beer {
  static names = [
    'ninteennintynine',
    'badenbadenstout',
    'bambergrauchbier',
    'dadobelgianale',
    'demoiselle',
    'eisenbahndunkel',
    'eisenbahnkoelsch',
    'eisenbahnpaleale',
    'xingublackbeer',
    'brahma',
  ]

  static all() {
    return Promise.all(Beer.names.map(name =>
      fetch(`http://prost.herokuapp.com/api/v1/beer/${name}`, {
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
      })
      .then(result => result.json())
    ))
  }
}
