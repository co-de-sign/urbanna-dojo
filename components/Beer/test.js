import React from 'react'
import Beer from '../Beer'
import renderer from 'react-test-renderer'

describe('Beer', () => {
  let beer

  beforeEach(() => {
    const testData = {
      name: 'test beer',
      alcohol: .3,
      type: ['lager'],
    }

    beer = renderer.create(<Beer beer={testData} />)
  })

  it('renders correctly', () => {
    expect(beer).toMatchSnapshot()
  })
})
