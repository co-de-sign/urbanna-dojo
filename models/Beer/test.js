import MockAsyncStorage from '../../helpers/MockAsyncStorage'

const mockStorage = new MockAsyncStorage()
jest.setMock('AsyncStorage', mockStorage)

import Beer from '.'

describe('Beer', () => {
  describe('#all', () => {
    const beerSample = {
      name: 'fake',
      alcohol: 1,
    }

    beforeAll(() => {
      global.fetch = require('jest-fetch-mock')
    })

    beforeEach(() => {
      fetch.mockResponse(JSON.stringify(beerSample))
    })

    afterEach(() => {
      fetch.resetMocks()
      mockStorage.clear()
    })

    it('returns all beers', async () => {
      const allBeers = await Beer.all()

      expect(allBeers.length).toBe(10)
    })

    it('returns the beer instance', async () => {
      const allBeers = await Beer.all()

      expect(allBeers[0]).toEqual(beerSample)
    })

    it('caches the beers after first call', async () => {
      let allBeers = await Beer.all()

      expect(fetch).toHaveBeenCalledTimes(10)

      allBeers = await Beer.all()

      expect(fetch).toHaveBeenCalledTimes(10)
    })
  })
})
