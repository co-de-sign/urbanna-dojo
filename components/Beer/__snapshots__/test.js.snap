// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": 244,
      "marginTop": 32,
      "width": 256,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test beer
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    0.3
    %
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    lager
  </Text>
</View>
`;
